projects/packages/ip/tests/php/test-utils.php

```diff
@@ -305,4 +305,45 @@ public function test_ip_address_is_in_range() {
 		$this->assertFalse( Utils::ip_address_is_in_range( $out_range_ip, $range_low, $range_high ) );
 	}
 
+	/**
+	 * Test `get_ip_addresses_from_string`.
+	 * Covers IPv4 and IPv6 addresses, including ranges, concatenated with various delimiters.
+	 *
+	 * @covers ::get_ip_addresses_from_string
+	 */
+	public function test_get_ip_addresses_from_string() {
+		$string = '';
+
+		$delimiters       = array( "\n", ',', ';', ' ' );
+		$delimiters_index = 0;
+
+		$ips = array(
+			// IPv4
+			'1.1.1.1',
+			'2.2.2.2',
+			'3.3.3.3',
+			'4.4.4.4',
+			'5.5.5.5-6.6.6.6',
+			// IPv6
+			'2001:db8::1',
+			'2001:db8::2',
+			'2001:db8::3',
+			'2001:db8::4',
+			'2001:db8::5-2001:db8::6',
+		);
+
+		// Generate a string that includes all IPs and delimiters.
+		foreach ( $ips as $ips_index => $ip ) {
+			$string .= $ip;
+
+			$is_last_loop = $ips_index === count( $ips ) - 1;
+			if ( ! $is_last_loop ) {
+				$string          .= $delimiters[ $delimiters_index ];
+				$delimiters_index = count( $delimiters ) === $delimiters_index + 1 ? 0 : $delimiters_index + 1;
+			}
+		}
```

Comment:
I'd certainly prefer if our expectation was hardcoded here, because this makes it pretty hard to understand what exactly we are expecting.

Reply:
Thanks for checking this out Kolja! Good point - it is much more understandable by hardcoding it: https://github.com/Automattic/jetpack/pull/29131/commits/677b08d52a833f20edcff0a59e3cb5feda4c2f3d